editor/plugins/script_editor_plugin.cpp

```diff
@@ -3575,89 +3575,33 @@ void ScriptEditor::_on_replace_in_files_requested(String text) {
 	find_in_files_dialog->popup_centered();
 }
 
-void ScriptEditor::_on_find_in_files_result_selected(String fpath, int line_number, int begin, int end) {
-	if (ResourceLoader::exists(fpath)) {
-		Ref<Resource> res = ResourceLoader::load(fpath);
+void ScriptEditor::_on_find_in_files_result_selected(String relative_path, String absolute_path, int line_number, int begin, int end) {
+	if (ResourceLoader::exists(absolute_path)) {
+		Ref<Resource> res = ResourceLoader::load(absolute_path);
 
-		if (fpath.get_extension() == "gdshader") {
+		if (absolute_path.get_extension() == "gdshader") {
 			ShaderEditorPlugin *shader_editor = Object::cast_to<ShaderEditorPlugin>(EditorNode::get_singleton()->get_editor_data().get_editor("Shader"));
 			shader_editor->edit(res.ptr());
 			shader_editor->make_visible(true);
 			shader_editor->get_shader_editor(res)->goto_line_selection(line_number - 1, begin, end);
 			return;
-		} else if (fpath.get_extension() == "tscn") {
-			Ref<FileAccess> f = FileAccess::open(fpath, FileAccess::READ);
-			bool is_script_found = false;
+		} else if (absolute_path.get_extension() == "tscn") {
+			EditorNode::get_singleton()->load_scene(absolute_path);
```

Comment:
I believe this whole exercise that you've replaced existed for performance reason, to avoid fully parsing entire scenes. `EditorNode::load_scene` does the regular `ResourceLoader::load` on top of many other things. I think this means we're getting worse performance for scenes which aren't fully loaded before you do the search and select a result, no?

Reply:
The previous version called both `EditorNode::load_scene` and `ResourceLoader::load` based on essentially the same set of conditions (absolute path has the extension "tscn" and it's a valid script).
Keep in mind `ScriptEditor::_on_find_in_files_result_selected` is only called when a result of the Find In Files "process" is selected, which means the user is actively selecting that search result and they want to edit it, so performance isn't paramount.